pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/assignment/instance/InstanceAssignmentDriver.java

```diff
@@ -55,19 +58,36 @@ public InstanceAssignmentDriver(TableConfig tableConfig) {
   public InstancePartitions assignInstances(InstancePartitionsType instancePartitionsType,
       List<InstanceConfig> instanceConfigs, @Nullable InstancePartitions existingInstancePartitions) {
     String tableNameWithType = _tableConfig.getTableName();
-    LOGGER.info("Starting {} instance assignment for table: {}", instancePartitionsType, tableNameWithType);
+    Preconditions.checkState(!TableConfigUtils.isTableInGroup(_tableConfig));
```

Comment:
Are there any concerns with making ZK propertyStore a member of `InstanceAssignmentDriver`? That would allow to get rid of this Precondition here and return the pre-computed instance-partitions by doing a look-up on the propertyStore here.

Otherwise we will have to do a check whenever we call this method to see if the table is part of a table-group and handle that case every time.